### ISSUE TITLE
ENH: add `split`

### ISSUE THREAD

**Issue by lucascolley** [data-apis/array-api-extra#321]
breaking this out from https://github.com/data-apis/array-api-extra/issues/97

**Comment by lucascolley** [comment]
returning a `tuple` may be nicer than returning a `list` I suppose. The main work seems to be in deciding which of the following is preferred:

> If indices_or_sections is a 1-D array of sorted integers, the entries indicate where along axis the array is split. 

vs.

> If split_size_or_sections is a list, then [tensor](https://docs.pytorch.org/docs/2.13/generated/torch.tensor.html#torch.tensor) will be split into len(split_size_or_sections) chunks with sizes in dim according to split_size_or_sections.
